File: Basic Programming/Dimik oj/Python3/প্রোগ্রামিং সমস্যা 38 - [৫২ সমস্যা বই] হীরক রাজ্য.py
#                      Name : Arif Khan
#                      Judge: Dimik Oj
#                      University: Primeasia University
#                      problem: প্রোগ্রামিং সমস্যা 38 - [৫২ সমস্যা বই] হীরক রাজ্য
#                      Difficulty: Easy
#                      Time Complexity: 0.3333 সেকেন্ড
#                      Problem Link: https://dimikoj.com/problems/38/herok-rajjo-by
#

for _ in range(int(input())):
    a, b = map(int, input().split())
    for i in range(1, a+1):
        print(*[b]*i)
    for i in range(a-1, 0, -1):
        print(*[b]*i)
    print()

File: Basic Programming/Dimik oj/C++/প্রোগ্রামিং সমস্যা 38 - [৫২ সমস্যা বই] হীরক রাজ্য.cpp
/*                    Name : Arif Khan
                      Judge: Dimik Oj
                      University: Primeasia University
                      problem: প্রোগ্রামিং সমস্যা 38 - [৫২ সমস্যা বই] হীরক রাজ্য
                      Difficulty: Easy
                      Time Complexity: 0.2326 সেকেন্ড
                      Problem Link: https://dimikoj.com/problems/38/herok-rajjo-by
*/

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, a, b, k;
	cin >> n;
	while(n--) {
	    cin >> a >> b;
	     k = a-1;
	    for(int i = 1; i < a*2; i++) {
	        for(int j = 1; i <= a and j <= i; j++) {
	            if(j != 1) cout << " ";
	            cout << b;
	        }
	        for(int j = 1; i > a and j <= k; j++) {
	            if(j != 1) cout << " ";
	            cout << b;
	        }
	        if(i > a) k--;
	        cout << endl;
	    }
	    cout << endl;
	}
	return 0;
}